freqtrade/freqtradebot.py

```diff
@@ -1374,6 +1375,15 @@ def _check_and_execute_exit(
         for should_exit in exits:
             if should_exit.exit_flag:
                 exit_tag1 = exit_tag if should_exit.exit_type == ExitType.EXIT_SIGNAL else None
+                if trade.has_open_orders:
+                    if prev_eval := self._exit_reason_cache.get(
+                        f"{trade.pair}_{trade.id}_{exit_tag1 or should_exit.exit_reason}", None
+                    ):
+                        logger.debug(
+                            f"Exit reason already seen this candle, first seen at {prev_eval}"
```

Comment:
As there is no element related to the current candle like the close date, what make you say that the cache is kept for the candle and not the whole trade and exit tag or reason?

Reply:
the cache is a PeriodicCache ... it'll expire as defined in `__init__()` - which is `timeframe_to_seconds(self.strategy.timeframe)` - or 300 for a 5m strategy.

Reply:
Good, got it

Reply:
the only slight downside is that a bot restart within a candle with an open order will trigger a recreation of the order if the signal is still there (as the cache resets)

Reply:
Yes but it is not a big deal as It will reset the order only once and a bot restart is supposed to be a rare event. The fix seem good even for this case.

Reply:
i've also tried to move it to this location as it'll not interfere with "intended" changes (replace-order or trade-adjustments) (and doing all of this in `handle_similar_open_order()` was terribly verbose as the info message below the changed lines still triggered :laughing: